### PR TITLE
Scenario name support -  Generate only matching feature files when <scenarioNames> is specified

### LINKED ISSUE #127: Group scenarios whose names match REGEXP together
When `parallelizationMode = scenarios`, group scenarios whose names match `REGEXP` together, in  the same runner file if possible (inspired by `cucumber.options="--name REGEXP"`).

**Use Case:**
I have a scenario where different types of users need to be tested for different capabilities (capabilities can be anything, for example, presence or absence of UI elements depending on the logged in user). Login is a very expensive operation which involves multiple steps and can take up to 30 seconds. So I would like to minimize the number of logins as much as possible. The scenario outline looks like this:
```
Scenario Outline: Test "<capability>" of "<user>"
    When I log in as a "<user>"
    Then I can test "<capability>"

    Scenarios:
      | user    | capability    |
      | user1   | cap1          |
      | user1   | cap2          |
      | user1   | cap3          |
      | user2   | cap4          |
      | user2   | cap5          |
      | user3   | cap6          |
      | user3   | cap7          |
```
With `desiredNumberOfRunners` set to 3, what I see now is:
```
runnerFile1:
    Test "cap1" of "user1"
    Test "cap4" of "user2"
    Test "cap7" of "user3"
runnerFile2:
    Test "cap2" of "user1"
    Test "cap5" of "user2"
runnerFile3:
    Test "cap3" of "user1"
    Test "cap6" of "user3"
```
This will lead to multiple logins in each runner.
What I would like to do is to be able to specify a `REGEXP` , in this case `user\d`, and see the scenarios grouped as below:
```
runnerFile1:
    Test "cap1" of "user1"
    Test "cap2" of "user1"
    Test "cap3" of "user1"
runnerFile2:
    Test "cap4" of "user2"
    Test "cap5" of "user2"
runnerFile3:
    Test "cap6" of "user3"
    Test "cap7" of "user3"
```
If this seems to be something worth doing, I would also like to propose an extension of the above where the user can specify a list of `REGEXP`, say `regex1, regex2, regex3`, and `desiredNumberOfRunners` automatically gets set to the number of items in the list (in this case, 3). The scenarios matching each `REGEXP` can then be moved to separate runner files.

### PR DESCRIPTION
**Pull request by josepj**
This PR resolves #127.

### CODE REVIEW

**Comment by laxersaz** [comment]
I hope I can check this out this week. There's a lot of stuff going on at the moment.

File: plugin-code/src/main/java/com/trivago/gherkin/GherkinDocumentParser.java [hunk 1]
```diff
@@ -353,7 +346,35 @@ private final boolean scenarioShouldBeIncluded(final List<String>... sourceTagsL
...
+    public int matchScenarioWithScenarioNames(String stringToMatch) {
+        List<String> scenarioNames = propertyManager.getScenarioNames();
+        int matchIndex = -1;
+
+        if (scenarioNames == null || scenarioNames.isEmpty()) {
+            return 0;
+        }
+
+        for (String scenarioName : scenarioNames) {
+            String regex = "(Scenario|Example):.+" + scenarioName;
```
Comment (laxersaz) [inline plugin-code/src/main/java/com/trivago/gherkin/GherkinDocumentParser.java]: This will not work with non-english scenarios/examples.

Comment (josepj) [reply plugin-code/src/main/java/com/trivago/gherkin/GherkinDocumentParser.java]: Completely forgot about the non-english part. Sorry! I will look into this.

Comment (josepj) [reply plugin-code/src/main/java/com/trivago/gherkin/GherkinDocumentParser.java]: I have checked in the changes. Please take a look.

**Comment by laxersaz** [comment]
@josepj I released version 1.5.3 including your scenario name change - might take some time until it is available on Maven Central.
Thanks again for your great work!

**Pull request merged**